File: frontend/src/types/ai-question-candidate.ts
export type AiQuestionCandidateStatus = 0 | 1 | 2 | 3 | 4 | 5

export interface AiQuestionCandidate {
  id: number
  title: string
  normalizedTitle: string
  answer: string | null
  difficulty: number | null
  categoryId: number | null
  tagIds: number[]
  direction: string
  sourceSessionId: number
  sourceMessageId: number
  duplicateCount: number
  status: AiQuestionCandidateStatus
  errorMessage: string | null
  questionId: number | null
  createTime: string
  updateTime: string
}

export interface AiQuestionCandidateQuery {
  pageNum: number
  pageSize: number
  status?: AiQuestionCandidateStatus
  direction?: string
  keyword?: string
}

export interface AiQuestionApprovePayload {
  title: string
  content?: string
  answer: string
  difficulty: number
  categoryId: number
  tagIds: number[]
}

File: src/main/java/com/zhimian/model/entity/AiQuestionCandidate.java
package com.zhimian.model.entity;

import lombok.Data;

import java.time.LocalDateTime;

@Data
public class AiQuestionCandidate {
    private Long id;
    private String title;
    private String normalizedTitle;
    private String fingerprint;
    private String answer;
    private Integer difficulty;
    private Long categoryId;

    // 数据库列 tag_ids 是 JSON。Mapper 查询时把它转成字符串，Service 再用 ObjectMapper 解析。
    private String tagIdsJson;
    private String direction;
    private Long sourceSessionId;
    private Long sourceMessageId;
    private Integer duplicateCount;
    private Integer status;
    private String errorMessage;
    private Long questionId;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
}
